src/pages/useGapsList.ts

```diff
@@ -43,6 +43,13 @@ export const useGapsList = (
 ): HourlyData[] => {
   const [hourlyData, setHourlyData] = useState<HourlyData[]>([])
 
+  // Cleanup function to reset the state when the component is unmounted
+  useEffect(() => {
+    return () => {
+      setHourlyData([])
+    }
+  }, [lineRef, operatorRef, fromDate, toDate, sortingMode])
+
   useEffect(() => {
     const fetchData = async () => {
       try {
```

Comment:
wdyt?
```suggestion
  useEffect(() => {
    const fetchData = async () => {
      try {
        setHourlyData([])
```

Reply:
While they will both work the same in regards of returning an empty array initially before data is fetched, the main benefit of having the cleanup is to avoid potential memory leaks or stale state issues down the line.

So both will work, I can remove the cleanup and use your solution if you like it better. Let me know!

Reply:
you're right, having a cleanup function is better approach.
shouldn't we add the cleanup to the existing `useEffect`?

By the way, for the long term I believe we should consider moving to `useQuery` in the future. That's why I prefer to keep all query-related logic in the same place (when possible)

Reply:
keeping everything in the same useEffect is a better approach, I guess I seperated them for my own brain lol.
Fixed and pushed!